File: src/main/java/unificado/core/service/cuentamovimientounificado/repository/CuentaMovimientoUnificadoRepository.java
package unificado.core.service.cuentamovimientounificado.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import unificado.core.service.cuentamovimientounificado.model.CuentaMovimientoUnificado;

import java.time.OffsetDateTime;
import java.util.Optional;

public interface CuentaMovimientoUnificadoRepository extends JpaRepository<CuentaMovimientoUnificado, Long> {

    Optional<CuentaMovimientoUnificado> findByNegocioIdAndNumeroCuentaAndDesdeAndHastaAndDebita(Integer negocioId, Long numeroCuenta, OffsetDateTime desde, OffsetDateTime hasta, Byte debita);

}

File: src/main/java/unificado/api/rest/kotlin/repository/CuentaMovimientoUnificadoRepository.kt
package unificado.api.rest.kotlin.repository

import org.springframework.data.jpa.repository.JpaRepository
import unificado.api.rest.kotlin.model.CuentaMovimientoUnificado
import java.time.OffsetDateTime
import java.util.Optional

interface CuentaMovimientoUnificadoRepository : JpaRepository<CuentaMovimientoUnificado, Long> {

    fun findByNegocioIdAndNumeroCuentaAndDesdeAndHastaAndDebita(negocioId: Int, numeroCuenta: Long, desde: OffsetDateTime, hasta: OffsetDateTime, debita: Byte): Optional<CuentaMovimientoUnificado?>?

}